### PR TITLE
push talent grid info to perks on item

### PR DESCRIPTION
**Pull request by kyleshay**
just started playing around with the talent grid stuff since i kept forgetting what things i had on each item.

it would be cool to show the full grid (including nodes that aren't chosen) but we need more screen real-estate for that.

but now we could at least see things like the scope: 

screen shot 2015-12-12 at 4 10 59 pm

and items selected in columns

screen shot 2015-12-12 at 4 04 18 pm

### CODE REVIEW

**Comment by kyleshay** [comment]
between `item.nodes` and the talent grid information, we can easily build the grid in the correct order to match the game. it's just the matter of moving that out. this at least gets us looping over the grid to pull out 'activated' things to display.

maybe it's a bit much though?

**Comment by bhollis** [comment]
Kyle, would you like me to merge this for the next beta? I like showing the scope and selected perk. A rebase on `dev` would help :-)

**Comment by kyleshay** [comment]
sounds good @bhollis 

i need to make some minor changes, i'll post here when it's good to go.

**Comment by kyleshay** [comment]
alright, good to go. we're just showing the active perks for now. can build a full talent grid in if we really wanted, am just currently filtering out on the `isActive` property. everything is sorted on the `nodeHash` property, that way they appear in the same order as in game (scope first,... etc)

weapon:

screen shot 2016-01-21 at 1 25 24 pm

armor: 

screen shot 2016-01-21 at 1 25 09 pm

**Comment by Mulchman** [comment]
This'll be awesome to have for those of us with duplicate weapons but
different perks.

**Comment by kyleshay** [comment]
yup! eventually i want a full talent grid, but that might be something that has to wait until v4 for. the info box would get to be rather large otherwise.

in v4 i think we're considering having a panel on the side with additional information. a talent grid would be especially needed if bungie ever let us update perks through the API.

**Comment by bhollis** [comment]
This is great. I actually like not having the full talent grid for now (until we can get a better display). Not sure about the side-bar idea - it means your mouse has to travel further from where you click.

File: app/scripts/services/dimStoreService.factory.js [hunk 1]
```diff
@@ -476,20 +489,47 @@
...
+        });
+        // other useful information about the item (this has info about reforge/etc)
+        // _.each(talents.independentNodeIndexes, function(set) {
+
+        createdItem.hasReforgeNode = !_.isEmpty(reforgeNodes);
+      }
+
+      // remove inactive perks, with this we actually lose passive perks (like exotic perks)
+      createdItem.perks = _.filter(createdItem.perks, function(perk) {
+        return perk.isActive;
```
Comment (bhollis) [inline app/scripts/services/dimStoreService.factory.js]: Tip for later: I think you could write this as:

```javascript
createdItems.perks = _.filter(createdItem.perks, 'isActive');
```

If you want to keep using the full function predicate, built-in `filter` might be nicer:

```javascript
createdItem.perks = createdItem.perks.filter(function(perk) { return perk.isActive; });
```

Same for `_.each`, etc. Modern JS has a lot of Underscore built-in.

**Comment by kyleshay** [comment]
thanks for the heads up, will do in the future ^_^. just figured we were using _ everywhere so i was just staying consistent. 

cheers!

**Comment by bhollis** [comment]
Oh, that could be cool.

**Comment by bhollis** [comment]
(and don't sweat the style - we do different things everywhere, and it doesn't matter)

**Comment by bhollis** [comment]
Hmm, after merging, I'm having a couple problems with this change:

1. I don't see any icons for these new perks. There's an icon URL, but it's a 404 on Bungie.net. On Bungie's site, they have a different URL for these perks!
2. The nodes are not arranged according to the order they are in the game.

**Comment by bhollis** [comment]
Well, I solved #1 by just updating the manifest files ;-)

**Comment by bhollis** [comment]
And I fixed the sorting by sorting on `column` instead of `nodeHash`.

**Comment by kyleshay** [comment]
yikes, thanks for changing what it sorts on. should have double checked that. column makes sense.

**Pull request merged**